src/App.tsx

```diff
@@ -17,18 +98,34 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onStatusChange={handleStatusChange}
+                onSearchChange={handleSearchChange}
+                onClearSearch={handleClearSearch}
+                searchQuery={searchQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && <Loader />}
+
+              <TodoList
+                todos={getFilteredTodos()}
+                onSelectTodo={handleSelectTodo}
+                selectedTodoId={selectedTodo?.id}
+              />
             </div>
           </div>
         </div>
       </div>
-
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal
+          todo={selectedTodo}
+          user={user}
+          isLoading={isLoading}
+          onClose={handleCloseModal}
+        />
```

Comment:
Verify that the `TodoModal` component accepts the props `todo`, `user`, `isLoading`, and `onClose` as expected. Ensure these props are correctly typed and implemented in the `TodoModal` component.